action/action_deserializer.go

```diff
@@ -6,15 +6,31 @@
 
 package action
 
-import "github.com/iotexproject/iotex-proto/golang/iotextypes"
+import (
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+)
 
 // Deserializer de-serializes an action
 type Deserializer struct {
+	withChainID bool
 }
 
 // ActionToSealedEnvelope converts protobuf to SealedEnvelope
 func (ad *Deserializer) ActionToSealedEnvelope(pbAct *iotextypes.Action) (SealedEnvelope, error) {
-	var selp SealedEnvelope
-	err := selp.LoadProto(pbAct)
+	var (
+		selp SealedEnvelope
+		err  error
+	)
+	if ad.withChainID {
+		err = selp.LoadProtoWithChainID(pbAct)
+	} else {
+		err = selp.LoadProto(pbAct)
+	}
 	return selp, err
 }
+
+// WithChainID sets whether or not to use chainID
+func (ad *Deserializer) WithChainID(with bool) *Deserializer {
```

Comment:
The interface looks strange. `WithChainID(int)` would be better.

Reply:
I replied in an earlier comment, it is 2 diff styles:
```
d := &Deserializer{}
if ctx.xxxFlag {
    d.WithChainID()
}
d.FromBlockProto()
```
vs.
```
(&Deserializer{}).WithChainID(ctx.xxxFlag).FromBlockProto()
```
also this code will be reverted after 180 HF, so would like to keep it simple